daft/io/partitioning.py

```diff
@@ -0,0 +1,304 @@
+
+        Examples:
+            >>> import daft.io.partitioning PartitionTransform
+            >>> transform = PartitionTransform.year()
+            >>> assert transform.is_year()
+        """
+        return self._partition_transform.is_year()
+
+    def is_month(self) -> bool:
+        """Checks if this is a month partition transform.
+
+        Returns:
+            bool: True if this is a month transform.
+
+        Examples:
+            >>> import daft.io.partitioning PartitionTransform
+            >>> transform = PartitionTransform.month()
+            >>> assert transform.is_month()
+        """
+        return self._partition_transform.is_month()
+
+    def is_day(self) -> bool:
+        """Checks if this is a day partition transform.
+
+        Returns:
+            bool: True if this is a day transform.
+
+        Examples:
+            >>> import daft.io.partitioning PartitionTransform
+            >>> transform = PartitionTransform.day()
+            >>> assert transform.is_day()
+        """
+        return self._partition_transform.is_day()
+
+    def is_hour(self) -> bool:
+        """Checks if this is an hour partition transform.
+
+        Returns:
+            bool: True if this is an hour transform.
+
+        Examples:
+            >>> import daft.io.partitioning PartitionTransform
+            >>> transform = PartitionTransform.hour()
+            >>> assert transform.is_hour()
+        """
+        return self._partition_transform.is_hour()
+
+    def is_iceberg_bucket(self) -> bool:
+        """Checks if this is an Iceberg bucket partition transform.
+
+        Returns:
+            bool: True if this is an Iceberg bucket transform.
+
+        Examples:
+            >>> import daft.io.partitioning PartitionTransform
+            >>> transform = PartitionTransform.iceberg_bucket(10)
+            >>> assert transform.is_iceberg_bucket()
+        """
+        return self._partition_transform.is_iceberg_bucket()
+
+    def is_iceberg_truncate(self) -> bool:
+        """Checks if this is an Iceberg truncate partition transform.
+
+        Returns:
+            bool: True if this is an Iceberg truncate transform.
+
+        Examples:
+            >>> import daft.io.partitioning PartitionTransform
+            >>> transform = PartitionTransform.iceberg_truncate(10)
+            >>> assert transform.is_iceberg_truncate()
+        """
+        return self._partition_transform.is_iceberg_truncate()
+
+    def get_iceberg_bucket_n(self) -> int:
+        """Gets the number of buckets for an Iceberg bucket transform.
+
+        Returns:
+            int: Number of buckets.
+
+        Examples:
+            >>> import daft.io.partitioning PartitionTransform
+            >>> transform = PartitionTransform.iceberg_bucket(10)
+            >>> assert transform.get_iceberg_bucket_n() == 10
+        """
+        return self._partition_transform.get_iceberg_bucket_n()
+
+    def get_iceberg_truncate_w(self) -> int:
```

Comment:
to mirror what we did with `class DataType`, I suggest we rename these to `num_buckets` and `width`. 

for example,

```py
dtype = daft.datatype.fixed_size_list(daft.datatype.int8(), 2)

size = dtype.size()
```
instead of 
```py
size = dtype.fixed_size_list_size()
```